dali/pipeline/util/operator_impl_utils.h

```diff
@@ -51,6 +55,123 @@ class OpImplBase<CPUBackend> {
   }
 };
 
+
+namespace detail {
+
+template <typename T>
+void ReadArgConstant(std::vector<T> &out, const string &arg_name, const OpSpec &spec) {
+  out.resize(1);
+  out[0] = spec.GetArgument<T>(arg_name);
+}
+
+template <typename T>
+void ReadArgInput(std::vector<T> &out, const std::string &arg_name,
+                  const OpSpec &spec, const workspace_t<CPUBackend> &ws) {
+  const auto& arg_in = ws.ArgumentInput(arg_name);
+  auto arg_in_view = view<const float>(arg_in);
+  DALI_ENFORCE(is_uniform(arg_in_view.shape) && volume(arg_in_view.shape[0]) == 1,
+    make_string("``", arg_name, "`` must be a scalar or a 1D tensor with a single element"));
+  if (arg_in_view.shape.sample_dim() > 0) {
+    DALI_WARN_ONCE("Warning: \"", arg_name, "\""
+                   " expected a scalar but received a 1D tensor with a single "
+                   "element. Please use a scalar instead.");
+  }
+
+  auto nsamples = arg_in_view.size();
+  out.resize(nsamples);
+  for (int i = 0; i < nsamples; i++) {
+    out[i] = arg_in_view[i].data[0];
+  }
+}
+
+template <typename T>
+void ReadArgInput(std::vector<std::vector<T>> &out, const std::string &arg_name,
```

Comment:
This interface is very inefficient - it forces non-contiguous storage, avoidable copy and limits the dimensionality of per-sample values to just 1.